coldfront/plugins/qumulo/management/commands/check_billing_cycles.py

```diff
@@ -0,0 +1,126 @@
+import logging
+
+from coldfront.config.env import ENV
+from coldfront.core.allocation.models import (
+    Allocation,
+    AllocationAttribute,
+    AllocationAttributeType,
+)
+from coldfront.core.resource.models import Resource
+
+from coldfront.plugins.qumulo.utils.aces_manager import AcesManager
+from coldfront.plugins.qumulo.utils.qumulo_api import QumuloAPI
+from coldfront.plugins.qumulo.utils.acl_allocations import AclAllocations
+from coldfront.plugins.qumulo.utils.active_directory_api import ActiveDirectoryAPI
+from django.db.models import OuterRef, Subquery
+
+from qumulo.lib.request import RequestError
+
+import time
+from datetime import datetime
+
+from typing import Optional
+
+logger = logging.getLogger(__name__)
+
+
+def conditionally_update_billing_cycle_types(
+    allocation,
+    billing_attribute,
+    billing_cycle,
+    prepaid_expiration,
+    prepaid_billing_start,
+) -> None:
+    today = datetime.today().strftime("%Y-%m-%d")
+    service_rate_attribute = AllocationAttributeType.objects.get(name="service_rate")
+    if billing_cycle == "prepaid":
+        if prepaid_expiration == today or prepaid_expiration < today:
+            logger.warn(f"Changing {allocation} billing_cycle to monthly")
+            AllocationAttribute.objects.filter(
+                allocation=allocation,
+                allocation_attribute_type=billing_attribute,
+            ).update(value="monthly")
+    elif billing_cycle == "monthly":
+        if prepaid_billing_start == today:
+            logger.warn(f"Changing {allocation} billing_cycle to prepaid")
+            AllocationAttribute.objects.filter(
+                allocation=allocation,
+                allocation_attribute_type=billing_attribute,
+            ).update(value="prepaid")
+            AllocationAttribute.objects.filter(
+                allocation=allocation,
+                allocation_attribute_type=service_rate_attribute,
+            ).update(value="subscription")
+
+
+def calculate_prepaid_expiration(
+    allocation, bill_cycle, prepaid_months, prepaid_billing_start, prepaid_expiration
```

Comment:
@cdeanj2 for clarity, assign the data type for the signature variables. For example, `prepaid_months: str`